### PR TITLE
en: add document for optimistic mode sharding merge

### PR DESCRIPTION
**Pull request by ireneontheway**
### What is changed, added or deleted? (Required)

Add document for optimistic mode sharding merge
This PR aligns a part of #155. The rest part is aligned by @TomShawn .

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v2.0 changes for now)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-dm/pull/155
- Other reference link(s): The rest part https://github.com/pingcap/docs-dm/pull/253

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: en/feature-online-ddl-scheme.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ In the task configuration file, `online-ddl-scheme` is at the same level of `nam
 ## ********* Basic configuration *********
 name: test                      # The name of the task. Should be globally unique.
 task-mode: all                  # The task mode. Can be set to `full`/`incremental`/`all`.
-is-sharding: true               # Whether it is a task to merge shards.
+shard-mode: "pessimistic"       # This needs to be configured if it is a shard merge task. The "pessimistic mode" is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set to the "optimistic mode".
```
Comment (csuzhangxc) [inline en/feature-online-ddl-scheme.md:30]:
```suggestion
shard-mode: "pessimistic"       # This needs to be configured if it is a shard merge task. The "pessimistic" mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set to the "optimistic" mode.
```

File: en/shard-merge-best-practices.md [hunk 2]
```diff
@@ -6,19 +6,19 @@ aliases: ['/docs/tidb-data-migration/dev/shard-merge-best-practices/']
 
 # Best Practices of Data Migration in the Shard Merge Scenario
 
-This document describes the features and limitations of [TiDB Data Migration](https://github.com/pingcap/dm) (DM) in the shard merge scenario and provides a data migration best practice guide for your application.
+This document describes the features and limitations of [TiDB Data Migration](https://github.com/pingcap/dm) (DM) in the shard merge scenario and provides a data migration best practice guide for your application (se the default "pessimistic mode").
```
Comment (csuzhangxc) [inline en/shard-merge-best-practices.md:9]: What does `se` mean?

File: en/task-configuration-file-full.md [hunk 3]
```diff
@@ -24,8 +24,7 @@ The following is the task configuration file template which allows you to perfor
 ## ********* Basic configuration *********
 name: test                      # The name of the task. Should be globally unique.
 task-mode: all                  # The task mode. Can be set to `full`/`incremental`/`all`.
-is-sharding: true               # Whether it is a task to merge shards.
-shard-mode: "optimistic"        # The shard DDL replication mode. Can be set to ""/`pessimistic`/`optimistic`.
+shard-mode: "pessimistic"       # This needs to be configured if it is a shard merge task. The "pessimistic mode" is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set to the "optimistic mode".
```
Comment (csuzhangxc) [inline en/task-configuration-file-full.md:27]:
```suggestion
shard-mode: "pessimistic"       # This needs to be configured if it is a shard merge task. The "pessimistic" mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set to the "optimistic" mode.
```

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

File: en/shard-merge-best-practices.md (hunk 2 above)
Comment (ireneontheway) [reply en/shard-merge-best-practices.md]: I've fixed it.

**Review by csuzhangxc — COMMENTED** [review]
LGTM

File: en/feature-online-ddl-scheme.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ In the task configuration file, `online-ddl-scheme` is at the same level of `nam
 ## ********* Basic configuration *********
 name: test                      # The name of the task. Should be globally unique.
 task-mode: all                  # The task mode. Can be set to `full`/`incremental`/`all`.
-is-sharding: true               # Whether it is a task to merge shards.
+shard-mode: "pessimistic"       # This needs to be configured if it is a shard merge task. The "pessimistic" mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set to the "optimistic" mode.
```
Comment (yikeke) [inline en/feature-online-ddl-scheme.md:30]:
```suggestion
shard-mode: "pessimistic"       # You need to configure this if it is a shard merge task. The "pessimistic" mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set it to the "optimistic" mode.
```

File: en/key-features.md [hunk 5]
```diff
@@ -553,14 +553,14 @@ DM supports merging the DML and DDL data in the upstream MySQL/MariaDB sharded t
...
-Currently, the shard merge feature is supported only in limited scenarios. For details, refer to [Sharding DDL usage Restrictions](feature-shard-merge.md#restrictions).
+Currently, the shard merge feature is supported only in limited scenarios. For details, refer to [Sharding DDL usage Restrictions in the pessimistic mode](feature-shard-merge-pessimistic.md#restrictions) and [Sharding DDL usage Restrictions in the optimistic mode](feature-shard-merge-optimistic.md#restrictions).
 
 ### Parameter configuration
 
 Set `is-sharding` to `true` in the task configuration file:
 
 ```
-is-sharding: true
+shard-mode: "pessimistic" # This needs to be configured if it is a shard merge task. The "pessimistic mode" is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set to the "optimistic mode".
```
Comment (yikeke) [inline en/key-features.md:563]:
```suggestion
shard-mode: "pessimistic" # You need to configure this if it is a shard merge task. The "pessimistic" mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set it to the "optimistic" mode.
```

File: en/shard-merge-best-practices.md [hunk 6]
```diff
@@ -6,19 +6,19 @@ aliases: ['/docs/tidb-data-migration/dev/shard-merge-best-practices/']
 
 # Best Practices of Data Migration in the Shard Merge Scenario
 
-This document describes the features and limitations of [TiDB Data Migration](https://github.com/pingcap/dm) (DM) in the shard merge scenario and provides a data migration best practice guide for your application.
+This document describes the features and limitations of [TiDB Data Migration](https://github.com/pingcap/dm) (DM) in the shard merge scenario and provides a data migration best practice guide for your application (the default "pessimistic" mode).
```
Comment (yikeke) [inline en/shard-merge-best-practices.md:9]:
```suggestion
This document describes the features and limitations of [TiDB Data Migration](https://github.com/pingcap/dm) (DM) in the shard merge scenario and provides a data migration best practice guide for your application (the default "pessimistic" mode is used).
```

File: en/skip-or-replace-abnormal-sql-statements.md [hunk 7]
```diff
@@ -33,7 +33,7 @@ If you know in advance that an unsupported SQL statement is going to be replicat
 
 - `--sharding` is only used to preset the operation to the sharding group. You must preset it before executing the DDL statement and presetting it after executing the DDL is not allowed.
     - `--sharding` only supports presetting operations, and in this mode, you can only use `--sql-pattern` to match the binlog event.
-    - For the principles of replicating sharding DDL statements using DM, see [Merge and replicate data from sharded tables](feature-shard-merge.md#principles)
+    - For the principles of replicating sharding DDL statements using DM, see [Merge and replicate data from sharded tables in the pessimistic mode](feature-shard-merge-pessimistic.md#restrictions) and [Merge and replicate data from sharded tables in pessimistic mode in the optimistic mode](feature-shard-merge-optimistic.md#restrictions).
```
Comment (yikeke) [inline en/skip-or-replace-abnormal-sql-statements.md:36]:
```suggestion
    - For the principles of replicating sharding DDL statements using DM, see [Merge and replicate data from sharded tables in the pessimistic mode](feature-shard-merge-pessimistic.md#restrictions) and [Merge and replicate data from sharded tables in the optimistic mode](feature-shard-merge-optimistic.md#restrictions).
```

File: en/overview.md [hunk 8]
```diff
@@ -98,7 +98,7 @@ Before using the DM tool, note the following restrictions:
 
     - If conflict exists between sharded tables, solve the conflict by referring to [handling conflicts of auto-increment primary key](shard-merge-best-practices.md#handle-conflicts-of-auto-increment-primary-key). Otherwise, data replication is not supported. Conflicting data can cover each other and cause data loss.
 
-    - For other sharding restrictions, see [Sharding DDL usage restrictions](feature-shard-merge.md#restrictions).
+    - For other sharding restrictions, see [Sharding DDL usage Restrictions in pessimistic mode](feature-shard-merge-pessimistic.md#restrictions) and [Sharding DDL usage Restrictions in optimistic mode](feature-shard-merge-optimistic.md#restrictions).
```
Comment (yikeke) [inline en/overview.md:101]:
```suggestion
    - For other sharding restrictions, see [Sharding DDL usage Restrictions in the pessimistic mode](feature-shard-merge-pessimistic.md#restrictions) and [Sharding DDL usage Restrictions in the optimistic mode](feature-shard-merge-optimistic.md#restrictions).
```
To unify the usage

File: en/TOC.md [hunk 9]
```diff
@@ -15,7 +15,10 @@
     - [Binlog Event Filter](key-features.md#binlog-event-filter)
     - [Replication Delay Monitoring](key-features.md#replication-delay-monitoring)
     - [Online DDL Scheme](feature-online-ddl-scheme.md)
-    - [Merge and Replicate Data from Sharded Tables](feature-shard-merge.md)
+    + Merge and Replicate Data from Sharded Tables
```
Comment (yikeke) [inline en/TOC.md:18]: We forgot to add en/feature-shard-merge-optimistic.md and en/feature-shard-merge-pessimistic.md to TOC.md. So I added them here. PTAL @csuzhangxc

File: zh/TOC.md [hunk 10]
```diff
@@ -12,7 +12,10 @@
     - [Block & Allow Lists](key-features.md#block--allow-table-lists)
     - [Binlog Event Filter](key-features.md#binlog-event-filter)
     - [Online-ddl-scheme](feature-online-ddl-scheme.md)
-    - [分库分表合并同步](feature-shard-merge.md)
+    + 分库分表合并同步
```
Comment (yikeke) [inline zh/TOC.md:15]: We forgot to add zh/feature-shard-merge-optimistic.md and zh/feature-shard-merge-pessimistic.md to TOC.md. So I added them here. PTAL @csuzhangxc

File: en/skip-or-replace-abnormal-sql-statements.md [hunk 11]
```diff
@@ -33,7 +33,7 @@ If you know in advance that an unsupported SQL statement is going to be replicat
 
 - `--sharding` is only used to preset the operation to the sharding group. You must preset it before executing the DDL statement and presetting it after executing the DDL is not allowed.
     - `--sharding` only supports presetting operations, and in this mode, you can only use `--sql-pattern` to match the binlog event.
-    - For the principles of replicating sharding DDL statements using DM, see [Merge and replicate data from sharded tables](feature-shard-merge.md#principles)
+    - For the principles of replicating sharding DDL statements using DM, see [Merge and replicate data from sharded tables in the pessimistic mode](feature-shard-merge-pessimistic.md#principles) and [Merge and replicate data from sharded tables in the optimistic mode](feature-shard-merge-optimistic.md#principles).
```
Comment (ireneontheway) [inline en/skip-or-replace-abnormal-sql-statements.md:36]:
```suggestion
    - For the principles of replicating sharding DDL statements using DM, see [Merge and replicate data from sharded tables in the pessimistic mode](feature-shard-merge-pessimistic.md#principles) and [Merge and replicate data from sharded tables in the optimistic mode](feature-shard-merge-optimistic.md#implementation-principle).
```

File: en/TOC.md (hunk 9 above)
Comment (csuzhangxc) [reply en/TOC.md:18]: LGTM, thanks

File: zh/TOC.md (hunk 10 above)
Comment (csuzhangxc) [reply zh/TOC.md:15]: LGTM, thanks

**Pull request merged**